backend/sirius-components-diagrams-layout/src/test/java/org/eclipse/sirius/components/diagrams/layout/incremental/NodeResizeTests.java

```diff
@@ -0,0 +1,156 @@
+package org.eclipse.sirius.components.diagrams.layout.incremental;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.core.api.IRepresentationDescriptionSearchService;
+import org.eclipse.sirius.components.diagrams.CustomizableProperties;
+import org.eclipse.sirius.components.diagrams.Diagram;
+import org.eclipse.sirius.components.diagrams.Node;
+import org.eclipse.sirius.components.diagrams.Position;
+import org.eclipse.sirius.components.diagrams.Size;
+import org.eclipse.sirius.components.diagrams.description.DiagramDescription;
+import org.eclipse.sirius.components.diagrams.events.IDiagramEvent;
+import org.eclipse.sirius.components.diagrams.events.ResizeEvent;
+import org.eclipse.sirius.components.diagrams.layout.ELKLayoutedDiagramProvider;
+import org.eclipse.sirius.components.diagrams.layout.IELKDiagramConverter;
+import org.eclipse.sirius.components.diagrams.layout.LayoutConfiguratorRegistry;
+import org.eclipse.sirius.components.diagrams.layout.LayoutService;
+import org.eclipse.sirius.components.diagrams.layout.incremental.provider.ImageSizeProvider;
+import org.eclipse.sirius.components.diagrams.layout.incremental.provider.NodeSizeProvider;
+import org.eclipse.sirius.components.diagrams.layout.services.DefaultTestDiagramDescriptionProvider;
+import org.eclipse.sirius.components.diagrams.layout.services.TestDiagramCreationService;
+import org.eclipse.sirius.components.diagrams.layout.services.TestLayoutObjectService;
+import org.eclipse.sirius.components.diagrams.tests.builder.JsonBasedEditingContext;
+import org.eclipse.sirius.components.diagrams.tests.builder.TestLayoutDiagramBuilder;
+import org.eclipse.sirius.components.representations.IRepresentationDescription;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests the manual resize of a node.
+ *
+ * @author rpage
+ */
+public class NodeResizeTests {
+    private TestLayoutObjectService objectService = new TestLayoutObjectService();
+
+    private DefaultTestDiagramDescriptionProvider defaultTestDiagramDescriptionProvider = new DefaultTestDiagramDescriptionProvider(this.objectService);
+
+    @Test
+    public void testResizeWithNoPositionDelta() {
+        Position positionDelta = Position.at(0, 0);
+        this.testResizeNode(positionDelta, Position.at(10, 10));
+    }
+
+    @Test
+    public void testResizeWithPositionDelta() {
+        // a positive position delta means the node has been resized from NW to the outside.
+        Position positionDelta = Position.at(10, 10);
+        this.testResizeNode(positionDelta, Position.at(0, 0));
+    }
```

Comment:
Could you please change the `testResizeNode` method to ask for both the position delta and the new size and the expected new position and new size in order to test resizing a node from the N, NE, E, SE, S, SW, W, NW